src/sentry/event_manager.py

```diff
@@ -1867,10 +1867,11 @@ def _create_group(project: Project, event: Event, **kwargs: Any) -> Group:
 
     # get severity score for use in alerting
     if features.has("projects:first-event-severity-calculation", event.project):
-        severity = _get_severity_score(event)
-        if severity is not None:  # Severity can be 0
-            group_data.setdefault("metadata", {})
-            group_data["metadata"]["severity"] = severity
+        severity, reason = _get_severity_score(event)
+
+        group_data.setdefault("metadata", {})
+        group_data["metadata"]["severity"] = severity
+        group_data["metadata"]["severity.reason"] = reason
```

Comment:
I'd go with `severity_reason`,or doing away with backwards compatibility and changing it to ```group_data["metadata"]["severity"] = {"value": severity, "reason": reason}```